src/pygeoprocessing/geoprocessing.py

```diff
@@ -1834,28 +1831,30 @@ def reclassify_raster(
             base_raster_path_band)
     raster_info = get_raster_info(base_raster_path_band[0])
     nodata = raster_info['nodata'][base_raster_path_band[1]-1]
-    value_map_copy = value_map.copy()
-    # possible that nodata value is not defined, so test for None first
-    # otherwise if nodata not predefined, remap it into the dictionary
-    if nodata is not None and nodata not in value_map_copy:
-        value_map_copy[nodata] = target_nodata
-    keys = sorted(numpy.array(list(value_map_copy.keys())))
-    values = numpy.array([value_map_copy[x] for x in keys])
+    keys = sorted(numpy.array(list(value_map.keys())))
+    values = numpy.array([value_map[x] for x in keys])
 
     def _map_dataset_to_value_op(original_values):
         """Convert a block of original values to the lookup values."""
+        out_array = numpy.full(original_values.shape, target_nodata)
+        if nodata is None:
+            valid_mask = numpy.full(original_values.shape, True)
+        else:
+            valid_mask = ~numpy.isclose(original_values, nodata)
+
         if values_required:
-            unique = numpy.unique(original_values)
-            has_map = numpy.in1d(unique, keys)
+            unique = numpy.unique(original_values[valid_mask])
+            has_map = numpy.isin(unique, keys)
             if not all(has_map):
                 missing_values = unique[~has_map]
                 raise ReclassificationMissingValuesError(
-                    f'The following {missing_values.size} raster values'
-                    f' {missing_values} from "{base_raster_path_band[0]}"'
-                    ' do not have corresponding entries in the ``value_map``:'
-                    f' {value_map}.', missing_values)
-        index = numpy.digitize(original_values.ravel(), keys, right=True)
-        return values[index].reshape(original_values.shape)
+                    f'The following {missing_values.size} raster values '
+                    f'{missing_values} from "{base_raster_path_band[0]}" '
+                    'do not have corresponding entries in the ``value_map``: '
+                    f'{value_map}.')
+        index = numpy.digitize(original_values[valid_mask], keys, right=True)
+        out_array[valid_mask] = values[index]
```

Comment:
Nice to have have the update to the more modern numpy APIs here that don't require 1-d arrays. 👍